src/frontend/screens/KillPasscode.tsx

```diff
@@ -0,0 +1,124 @@
+import * as React from "react";
+import { defineMessages, FormattedMessage } from "react-intl";
+import { StyleSheet, Switch, View } from "react-native";
+import { useNavigation } from "react-navigation-hooks";
+import { NavigationStackScreenComponent } from "react-navigation-stack";
+import { KILL_PASSCODE } from "../constants";
+import { SecurityContext } from "../context/SecurityContext";
+import { LIGHT_GREY } from "../lib/styles";
+import HeaderTitle from "../sharedComponents/HeaderTitle";
+import IconButton from "../sharedComponents/IconButton";
+import { BackIcon } from "../sharedComponents/icons";
+import Text from "../sharedComponents/Text";
+
+const m = defineMessages({
+  title: {
+    id: "screens.KillPasscode.title",
+    defaultMessage: "Kill Passcode",
+  },
+  body: {
+    id: "screens.KillPasscode.body",
+    defaultMessage:
+      "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur",
+  },
+  toggleMessage: {
+    id: "screens.KillPasscode.toggleMessage",
+    defaultMessage: "Use Kill Passcode",
+  },
+  instructions: {
+    id: "screens.KillPasscode.instructions",
+    defaultMessage: "Enter the code above to hide sensitive data in Mapeo",
+  },
+});
+
+export const KillPasscode: NavigationStackScreenComponent = () => {
+  const [state, dispatch] = React.useContext(SecurityContext);
+  const { navigate } = useNavigation();
+
+  React.useEffect(() => {
+    //If the user does not have a passcode set, they should NOT be able to see this screen
+    //Also, if the user is in killmode, we do not want the person who has apprehended the phone to see this screen!
+    if (!state.passcode || state.appMode === "kill") navigate("Security");
+  }, []);
+
+  return (
+    <View style={styles.container}>
+      <Text style={styles.title}>
+        <FormattedMessage {...m.title} />
+      </Text>
+      <Text>
+        <FormattedMessage {...m.body} />
+      </Text>
+
+      <View style={styles.switch}>
+        <Text>
+          <FormattedMessage {...m.toggleMessage} />
+        </Text>
+        <Switch
+          onChange={() => {
+            dispatch({ type: "killModeEnabled:toggle" });
+          }}
+          value={state.killModeEnabled}
+        />
```

Comment:
think we discussed using a checkbox for this (not sure if that was gonna be a follow up), but if not, just wanted to note that it's kind of hard to see the switch in it's off state (it just shows the knob but nothing else)

Reply:
We discussed using a checkbox instead of a switch for other thing, but the designs were never updated for this component, so we should check in with Sabella about that as well!

Reply:
Converted to checkbox